actionpack/lib/action_dispatch/testing/assertions/response.rb

```diff
@@ -82,6 +84,19 @@ def normalize_argument_to_redirection(fragment)
             handle._compute_redirect_to_location(@request, fragment)
           end
         end
+
+        def generate_response_message(type)
+          message = "Expected response to be a <#{type}>, but was"
+
+          if @response.redirection?
+            redirect_is = normalize_argument_to_redirection(@response.location)
+            message << " a redirect to <#{redirect_is}>"
```

Comment:
Obscuring the type of redirect that occurred seems unhelpful... if we `assert_response 301`, then we could be told "Expected response to be a <301>, but was a redirect to <..>", which sounds a lot like "error: no error" without an indication that the response was a 302.